pkg/grpc/databroker/sync.go

```diff
@@ -0,0 +1,110 @@
+package databroker
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/pomerium/pomerium/internal/log"
+	"github.com/pomerium/pomerium/pkg/protoutil"
+)
+
+// SyncRecords calls fn for every record using Sync.
+func SyncRecords[T any, TMessage interface {
```

Comment:
was this code from https://github.com/pomerium/pomerium/pull/5234/files? should we un-revert it instead?

Reply:
That was an iterator based version. I had originally intended it for this work, but then the code changed and I couldn't use it anymore. In particular the SyncLatest returns the server and latest record versions, so the stream has two sets of data within it. `iter.Seq2` is already kinda clumsy with errors, but it started getting really gnarly with those versions too.

It could maybe work with just the normal `Sync`, but the consistency of these two methods was nice.